### DISCUSSION TITLE
OpenAPI spec to HTTP request

### DISCUSSION THREAD

**Discussion by electriquo** [OAI/OpenAPI-Specification#3323] [Q&A]
OpenAPI specification describes URL path, the HTTP method, the body of the request (if needed), the parameters (if needed), the properties (if needed), etc.

For instance, for the OpenAPI specification below

```yaml
paths:
  "/stub":
    patch:
      requestBody:
        content:
          application/json:
            schema:
              type: object
              properties:
                name:
                  type: string
                  description: The name of the stub.
```

The corresponding HTTP request is for the `/stub` path, using `PATCH` HTTP method and the request body is a JSON `{"name":"some name"}`.

Is there any tool that reads OpenAPI specification file and issue a request per the specification or returns an HTTP request object (of some sort)?

**Comment by handrews** [comment]
Please take a look at some sites that list different tools:

* https://tools.openapis.org/
* https://openapi.tools/ 

We avoid recommending specific tools here in order to keep the specification neutral.
